concrete-core-fixture/src/fixture/lwe_ciphertext_consuming_retrieval.rs

```diff
@@ -0,0 +1,347 @@
+    LweCiphertextView: LweCiphertextEntity,
+    Maker: SynthesizesLweCiphertextView<Precision, LweCiphertextView>,
+{
+    type Parameters = LweCiphertextConsumingRetrievalParameters;
+    type RepetitionPrototypes = ();
+    type SamplePrototypes =
+        (<Maker as PrototypesLweCiphertext<Precision, LweCiphertextView::KeyDistribution>>::LweCiphertextProto,);
+    type PreExecutionContext = (LweCiphertextView,);
+    type PostExecutionContext = (Vec<Precision::Raw>,);
+    type Criteria = (Variance,);
+    type Outcome = (Vec<Precision::Raw>, Vec<Precision::Raw>);
+
+    fn generate_parameters_iterator() -> Box<dyn Iterator<Item = Self::Parameters>> {
+        Box::new(
+            vec![
+                LweCiphertextConsumingRetrievalParameters {
+                    lwe_dimension: LweDimension(1),
+                },
+                LweCiphertextConsumingRetrievalParameters {
+                    lwe_dimension: LweDimension(512),
+                },
+                LweCiphertextConsumingRetrievalParameters {
+                    lwe_dimension: LweDimension(751),
+                },
+            ]
+            .into_iter(),
+        )
+    }
+
+    fn generate_random_repetition_prototypes(
+        _parameters: &Self::Parameters,
+        _maker: &mut Maker,
+    ) -> Self::RepetitionPrototypes {
+    }
+
+    fn generate_random_sample_prototypes(
+        parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+    ) -> Self::SamplePrototypes {
+        let num_elements = parameters.lwe_dimension.to_lwe_size().0;
+        let proto_ciphertext_view =
+            maker.transform_raw_to_ciphertext(Precision::Raw::uniform_vec(num_elements));
+        (proto_ciphertext_view,)
+    }
+
+    fn prepare_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+    ) -> Self::PreExecutionContext {
+        let (proto_ciphertext_view,) = sample_proto;
+        (maker.synthesize_lwe_ciphertext_view(proto_ciphertext_view),)
+    }
+
+    fn execute_engine(
+        _parameters: &Self::Parameters,
+        engine: &mut Engine,
+        context: Self::PreExecutionContext,
+    ) -> Self::PostExecutionContext {
+        let (ciphertext_view,) = context;
+        let raw_ciphertext = unsafe {
+            engine
+                .consume_retrieve_lwe_ciphertext_unchecked(ciphertext_view)
+                .to_vec()
+        };
+        (raw_ciphertext,)
+    }
+
+    fn process_context(
+        _parameters: &Self::Parameters,
+        maker: &mut Maker,
+        _repetition_proto: &Self::RepetitionPrototypes,
+        sample_proto: &Self::SamplePrototypes,
+        context: Self::PostExecutionContext,
+    ) -> Self::Outcome {
+        let (proto_ciphertext_view,) = sample_proto;
+        let (raw_ciphertext,) = context;
```

Comment:
Again

Reply:
same

Reply:
this has changed in the last rework that will be pushed, resolving